config/tests/jobs/jobs_test.go

```diff
@@ -774,7 +774,89 @@ func checkScenarioArgs(jobName string, args []string) error {
 	}
 
 	if use_shared_build_in_args && build_in_args {
-		return fmt.Errorf("Job %s: --use-shared-build and --build cannot be combined", jobName)
+		return fmt.Errorf("job %s: --use-shared-build and --build cannot be combined", jobName)
+	}
+
+	if scenario != "kubernetes_e2e" {
+		return nil
+	}
+
+	if hasArg("--provider=gke", args) {
```

Comment:
it probably doesn't matter given how small the typical list of args is but I feel bad about how many times we loop over the args to check if an arg exists... I instinctively want to make a map and check that